File: Solutions/633. Sum of Square Numbers/633.java
class Solution {
    public boolean judgeSquareSum(int c) {
        long left = 0; 
        long right = (long)Math.sqrt(c) + 1;
        long total = 0;
        while(left <= right){
            total = (left * left) + (right * right);
            if(total == c){
                return true;
            }
            if(total < c){
                left++;
            }else{
                right--;
            }
        }
        return false;
    }
}

File: Solutions/633. Sum of Square Numbers/633.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        left, right = 0, int(sqrt(c))
        while left <= right:
            total = (left * left) + (right * right)
            if total == c:
                return True
            if total < c:
                left += 1
            else:
                right -= 1
        return False